### PR TITLE
Fix pulse circle opacity calculate error

### LINKED ISSUE #611: LocationComponent Pulse Graphical Issue
Upon updating to Maps SDK (for Android) **9.5.0**, I've noticed a visual peculiarity occurring when I enable `LocationComponent` pulsing. The issue is immediately apparent as the pulse does not smoothly "pulse" outward, but rather it pulses to its maximum diameter before flashing solid, and commences again. The bug is not present in Maps SDK (for Android) **9.4.0**.

Example that I've used as a foundation for implementing a pulsing `LocationComponent`:
https://docs.mapbox.com/android/maps/examples/location-component-custom-pulsing/

As a stop-gap solution, I've remained on Maps SDK (for Android) **9.4.0** where the bug is not present.

### PR DESCRIPTION
**Pull request by Chaoba**
Resolves #611

### CODE REVIEW

File: MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/location/LocationLayerController.java [hunk 1]
```diff
@@ -304,7 +304,7 @@ public void onNewAnimationValue(Float value) {
       public void onNewAnimationValue(Float newPulseRadiusValue) {
         Float newPulseOpacityValue = null;
         if (options.pulseFadeEnabled()) {
-          newPulseOpacityValue = (float) 1 - ((newPulseRadiusValue / 100) * 3);
+          newPulseOpacityValue =  1 - newPulseRadiusValue / options.pulseMaxRadius();
```
Comment (kiryldz) [inline MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/location/LocationLayerController.java:307]: nit: 1.0f will be clearer imho

**Review by kiryldz — APPROVED** [review]

File: MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/location/LocationLayerController.java [hunk 2]
```diff
@@ -304,7 +304,7 @@ public void onNewAnimationValue(Float value) {
       public void onNewAnimationValue(Float newPulseRadiusValue) {
         Float newPulseOpacityValue = null;
         if (options.pulseFadeEnabled()) {
-          newPulseOpacityValue = (float) 1 - ((newPulseRadiusValue / 100) * 3);
+          newPulseOpacityValue = 1.0f - newPulseRadiusValue / options.pulseMaxRadius();
```
Comment (alexshalamov) [inline MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/location/LocationLayerController.java:307]: Can `options.pulseMaxRadius()` be 0?

Comment (Chaoba) [reply MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/location/LocationLayerController.java:307]: Good catch. `newPulseRadiusValue / options.pulseMaxRadius()` can never be more than 1.0f, it's in [0,1] here.

**Review by alexshalamov — APPROVED** [review]

**Comment by bomturbo** [comment]
### Context

**Pull request merged**